src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -20,97 +20,105 @@
+        }
 
-            public Signer[] GetSigners()
+        private void Dispose_SmartContract()
+        {
+            timer.Dispose();
+            Session[] toBeDestroyed;
+            lock (sessions)
             {
-                return _signers;
+                toBeDestroyed = sessions.Values.ToArray();
+                sessions.Clear();
             }
+            foreach (Session session in toBeDestroyed)
+                session.Dispose();
+        }
 
-            public void SerializeUnsigned(BinaryWriter writer)
+        private void OnTimer(object state)
+        {
+            List<(Guid Id, Session Session)> toBeDestroyed = new();
+            lock (sessions)
             {
-                throw new NotImplementedException();
+                foreach (var (id, session) in sessions)
+                    if (DateTime.UtcNow >= session.StartTime + settings.SessionExpirationTime)
+                        toBeDestroyed.Add((id, session));
+                foreach (var (id, _) in toBeDestroyed)
+                    sessions.Remove(id);
             }
+            foreach (var (_, session) in toBeDestroyed)
+                session.Dispose();
         }
 
-        private JObject GetInvokeResult(byte[] script, Signers signers = null, bool useDiagnostic = false)
+        private JObject GetInvokeResult(byte[] script, Signer[] signers = null, Witness[] witnesses = null, bool useDiagnostic = false)
         {
-            Transaction tx = signers == null ? null : new Transaction
-            {
-                Signers = signers.GetSigners(),
-                Attributes = System.Array.Empty<TransactionAttribute>(),
-                Witnesses = signers.Witnesses,
-            };
-            using ApplicationEngine engine = ApplicationEngine.Run(script, system.StoreView, container: tx, settings: system.Settings, gas: settings.MaxGasInvoke, diagnostic: useDiagnostic ? new Diagnostic() : null);
             JObject json = new();
-            json["script"] = Convert.ToBase64String(script);
-            json["state"] = engine.State;
-            json["gasconsumed"] = engine.GasConsumed.ToString();
-            json["exception"] = GetExceptionMessage(engine.FaultException);
-            json["notifications"] = new JArray(engine.Notifications.Select(n =>
-              {
-                  var obj = new JObject();
-                  obj["eventname"] = n.EventName;
-                  obj["contract"] = n.ScriptHash.ToString();
-                  obj["state"] = ToJson(n.State, settings.MaxIteratorResultItems);
-                  return obj;
-              }));
-            if (useDiagnostic)
-            {
-                Diagnostic diagnostic = (Diagnostic)engine.Diagnostic;
-                json["diagnostics"] = new JObject()
+            Session session = new(system, script, signers, witnesses, settings.MaxGasInvoke, useDiagnostic ? new Diagnostic() : null);
```

Comment:
Do we always need to create it? Maybe it is better only if we send the parameter "cookie"

Reply:
If we don't need it, we will destroy it later.